src/Java.Interop/Java.Interop/JniTypeSignatureAttribute.cs

```diff
@@ -31,6 +32,9 @@ public      int         ArrayRank               {
 		}
 
 		public      bool        GenerateJavaPeer        {get; set;}
+
+		[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)]
+		public      Type?       InvokerType             {get; set;}
```

Comment:
The trimmer warnings might not require this, but should this also preserve interfaces? Just thinking generally, if it *needs to*.

In theory, that would make every interface preserve its interface implementations?

* Trimmer decides `IFoo` is used, and wants to keep the type
* `InvokerType=typeof(FooInvoker)` preserves the invoker
* `InvokerType=typeof(FooInvoker)` then could also preserve all interface implementations of `IFoo`?

Reply:
I think the TL;DR is that "as-is", the NativeAOT linker *cannot* be made to do what we want it to do, and I'm not sure if it'll "ever" be supported in some "reasonable" timeframe.

[Related](https://github.com/dotnet/java-interop/issues/1263#issuecomment-2517338914):

> I would also consider not using `All` unless we absolutely have to. `All` is kind of weird in that it will keep really everything about the type, so for example if the type implements an interface it will force all the methods on that interface to be kept (for everybody) and so on.

Part of the issue is that this `InvokerType` property is used for *both* abstract classes *and* interfaces.

The semantic we "need" is "preserve *all* method overrides/interface implementations."  We can't express that semantic right now (afaik).

Trying to "fake" the above semantic with "okay, just preserve everything on the type" means that we'll preserve *everything* on the type.  This is "fine" for interfaces (but see below), but is *not* fine for abstract classes.  Consider [`LauncherActivity`](https://learn.microsoft.com/en-us/dotnet/api/android.app.launcheractivity?view=net-android-34.0): as it's an Context+Activity subclass, it contains *dozens* of members through the inheritance chain.  We almost certainly do ***not*** need most of that infrastructure; we only need the infrastructure used by the method override!

Since, afaik, we can't actually get our desired semantic right now, and "preserve everything" looks like it would have terrible implications on app size when applied to abstract classes, I see only two solutions here:

  * `InterfaceInvokerType` vs. `AbstractClassInvokerType` properties!  Which is *possible*, but *ugh*.
  * Go with the least restrictive yet useful constraint (preserve constructors), and rely on "stuff elsewhere" to ensure things work and we get closer to our ideal semantic.

Additionally, "preserve everything" isn't even a useful constraint on interfaces!  It's *mostly* correct, for ***non-default instance methods***.  Pull in a default method, or static members, and we certainly *should* be able to trim those away!

Reply:
Yeah, I was just suggesting interfaces, not `All`, but yeah that doesn't make sense for abstract classes.